middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -1578,6 +1482,267 @@ let close_let_rec acc env ~function_declarations
+            Apply_cont_with_acc.create acc apply.continuation
+              ~args:(List.map BP.simple over_application_results)
+              ~dbg:apply_dbg
+          in
+          acc, Expr.create_apply_cont apply_cont_expr
+        in
+        Let_with_acc.create acc
+          (Bound_pattern.singleton
+             (Bound_var.create (Variable.create "unit") Name_mode.normal))
+          (Named.create_prim (Unary (End_region, Simple.var region)) apply_dbg)
+          ~body:call_return_continuation
+        |> Expr_with_acc.create_let
+      in
+      Let_cont_with_acc.build_non_recursive acc after_over_application
+        ~handler_params:over_application_results ~handler
+        ~body:(fun acc -> Expr_with_acc.create_apply acc over_application)
+        ~is_exn_handler:false
+  in
+  let body = full_call wrapper_cont in
+  let acc, both_applications =
+    Let_cont_with_acc.build_non_recursive acc wrapper_cont
+      ~handler_params:[BP.create returned_func K.With_subkind.any_value]
+      ~handler:perform_over_application ~body ~is_exn_handler:false
+  in
+  match needs_region with
+  | None -> acc, both_applications
+  | Some (region, _) ->
+    Let_with_acc.create acc
+      (Bound_pattern.singleton (Bound_var.create region Name_mode.normal))
+      (Named.create_prim (Nullary Begin_region) apply_dbg)
+      ~body:both_applications
+    |> Expr_with_acc.create_let
+
+type call_args_split =
+  | Exact of IR.simple list
+  | Partial_app of IR.simple list * Flambda_arity.With_subkinds.t
+  | Over_app of IR.simple list * IR.simple list
+
+let close_apply acc env (apply : IR.apply) : Acc.t * Expr_with_acc.t =
+  let callee = find_simple_from_id env apply.func in
+  let approx = Env.find_value_approximation env callee in
+  let code_info =
+    match approx with
+    | Closure_approximation (_, _, Code_present code) ->
+      Some
+        ( Code.params_arity code,
+          Code.is_tupled code,
+          Code.num_trailing_local_params code,
+          Code.contains_no_escaping_local_allocs code,
+          Code.result_arity code )
+    | Closure_approximation (_, _, Metadata_only metadata) ->
+      Some
+        ( Code_metadata.params_arity metadata,
+          Code_metadata.is_tupled metadata,
+          Code_metadata.num_trailing_local_params metadata,
+          Code_metadata.contains_no_escaping_local_allocs metadata,
+          Code_metadata.result_arity metadata )
+    | Value_unknown -> None
+    | _ ->
+      if Flambda_features.check_invariants ()
+      then
+        Misc.fatal_errorf
+          "Unexpected approximation for callee %a in [Closure_conversion], \
+           expected a closure approximation."
+          Simple.print callee
+      else None
+  in
+  match code_info with
+  | None -> close_exact_or_unknown_apply acc env apply None
+  | Some
+      ( arity,
+        is_tupled,
+        num_trailing_local_params,
+        contains_no_escaping_local_allocs,
+        result_arity ) -> (
+    let splited_args =
```

Comment:
`split_args` is better (split is an irregular verb). I think the form `splitted` is sometimes used too, but `splited` looks weird.

Reply:
"splitted" should never be used, always "split".